editors/code/src/ctx.ts

```diff
@@ -90,16 +90,12 @@ export async function sendRequestWithRetry<R>(
 ): Promise<R> {
     for (const delay of [2, 4, 6, 8, 10, null]) {
         try {
-            return await (token ? client.sendRequest(method, param, token) : client.sendRequest(method, param));
-        } catch (e) {
-            if (
-                e.code === lc.ErrorCodes.ContentModified &&
-                delay !== null
-            ) {
-                await sleep(10 * (1 << delay));
-                continue;
+            return await client.sendRequest(method, param, token);
+        } catch (err) {
+            if (delay === null || err.code !== lc.ErrorCodes.ContentModified) {
+                throw err;
             }
-            throw e;
+            await sleep(10 * (1 << delay));
```

Comment:
@matklad is this heuristic or something confirmed to be optimal?

Reply:
heuristic

Reply:
It's somewhat similar (minus a `rand`) to https://en.wikipedia.org/wiki/Exponential_backoff which works well in practice.